Expose common actor Inits in the map editor.

pchote opened the pull request:
This PR expands on #15551 by adding plumbing for trait-defined actor properties in the editor, and wiring up actor health, facing, turret facing and stance:

<img width="389" alt="screenshot 2018-12-03 at 11 40 05" src="https://user-images.githubusercontent.com/167819/49351875-56952900-f6f0-11e8-8f91-2f9e521b7cd5.png">

Fixes #15849.
Fixes #10818.

pchote committed:
Fix damagestates in the map editor.

pchote committed:
Remove legacy editor actor properties plumbing.

MustaphaTR reviewed (dismissed):
Works as described. Tho i'm not sure what to think about the order. Seems like they are dependent on Trait orders on yaml, but it would be way better imo for example if facing and turret facing where next to each other.

![resim](https://user-images.githubusercontent.com/7933210/49649984-e235ef00-fa3b-11e8-8b22-48a8a764f92a.png)

reaperrr commented:
Generally looks good, just two things I'd prefer to see changed/added (if not in this PR, then at least in a follow-up, but before a playtest):
- The order feels a bit arbitrary, personally I'd prefer Owner -> Stance -> Health -> Facing -> Turret.
- We currently still don't have any means to change an actor's position once it's been placed, it would be great if we could add that before branching off a playtest (not a blocker for this PR, though).

Adding fixup label for the order changes, feel free to reject it if there are technical reasons why this can't be done easily.

pchote committed:
Expose common actor Inits in the map editor.

pchote committed:
Simplify and fix panel positioning at different zooms.

pchote commented:
>The order feels a bit arbitrary, personally I'd prefer Owner -> Stance -> Health -> Facing -> Turret.

Done, and added plumbing that allows this to be overridden from yaml.

> We currently still don't have any means to change an actor's position once it's been placed

The preview rendering assumes that the preview never moves, so this is easier said than done.  I don't think this is likely to be fixed anytime soon.

I've also added a new commit that fixes the dialog position overlapping the actor at larger zooms.

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3ba0301b79a2cfef90352d13557d1fc2315a4171...00d68682bf355e107b67286e32fa530b8007b9b5
